pkg/controller/efs/filesystem/setup.go

```diff
@@ -67,11 +74,54 @@ func SetupFileSystem(mgr ctrl.Manager, o controller.Options) error {
 		Complete(r)
 }
 
-func isUpToDate(_ context.Context, cr *svcapitypes.FileSystem, obj *svcsdk.DescribeFileSystemsOutput) (bool, string, error) {
+type custom struct {
+	kube     client.Client
+	client   svcsdkapi.EFSAPI
+	external *external
+
+	cache struct {
+		backupPolicy bool
+	}
+}
+
+func (e *custom) cacheBackupPolicyHelper(filesystemId *string) error {
+	// Get BackupPolicy
+	//default is no backup
+	policy, err := e.client.DescribeBackupPolicy(&svcsdk.DescribeBackupPolicyInput{FileSystemId: filesystemId})
+	if err != nil {
+		var awsErr awserr.Error
+		if errors.As(err, &awsErr) && awsErr.Code() == "PolicyNotFound" {
+			e.cache.backupPolicy = false
+			return nil
+		}
+		return err
+	}
+
+	switch *policy.BackupPolicy.Status {
+	case "ENABLED", "ENABLING":
+		e.cache.backupPolicy = true
+	case "DISABLED", "DISABLING":
+		e.cache.backupPolicy = false
+	}
+	return nil
+}
+
+func (e *custom) isUpToDate(_ context.Context, cr *svcapitypes.FileSystem, obj *svcsdk.DescribeFileSystemsOutput) (bool, string, error) {
 	for _, res := range obj.FileSystems {
 		if pointer.Int64Value(cr.Spec.ForProvider.ProvisionedThroughputInMibps) != int64(aws.Float64Value(res.ProvisionedThroughputInMibps)) {
 			return false, "", nil
 		}
+		if !ptr.Equal(cr.Spec.ForProvider.ThroughputMode, res.ThroughputMode) {
+			return false, "", nil
+		}
+
+		err := e.cacheBackupPolicyHelper(res.FileSystemId)
+		if err != nil {
+			return false, "", err
```

Comment:
We should always `errors.Wrap` with a custom message to easily identify where the error comes from.